warp-core/src/main/scala/com/workday/warp/persistence/PersistenceAware.scala

```diff
@@ -49,13 +51,11 @@ trait PersistenceAware {
       * @tparam Row type of the row that we are referencing.
       * @return preexisting [[Row]], or newly inserted [[Row]].
       */
-    def findOrCreate[Table, Row](find: DBIO[Seq[Row]], create: DBIO[Row]): Row = {
+      // TODO make this more generic to accept an Option transparently.
+    def findOrCreate[Table, Row](find: DBIO[Iterable[Row]], create: DBIO[Row]): Row = {
```

Comment:
Wait, when you say "to accept an Option _transparently,_" does that mean this takes an Option now? I thought Option wasn't an Iterable

Reply:
no, this doesnt take a `DBIO[Option[T]]` now, a caller would need to do something like `d.map(_.toSeq)` to convert a `DBIO[Option[T]]` into a `DBIO[Seq[T]]`